typegen/strats/inline.py

```diff
@@ -101,67 +103,104 @@ def visit_FunctionDef(self, fdef: ast.FunctionDef):
             logger.debug(f"Applying return type hint '{ret_hint}' to '{fdef.name}'")
             fdef.returns = ast.Name(ret_hint)
 
+        self.generic_visit(fdef)
         return fdef
 
-    def visit_Assign(self, node: ast.Assign) -> ast.AST | list[ast.AST]:
+    def visit_Assign(self, node: ast.Assign) -> ast.Assign | ast.AnnAssign | list[ast.AST]:
         if not node.value:
             return node
 
-        logger.debug(f"Applying hints to '{ast.dump(node)}'")
+        target_names_with_nodes = list()
+        for target in node.targets:
+            target_names_with_nodes += self._extract_target_names_with_nodes(target)
+        if len(target_names_with_nodes) == 0:
+            return node
 
+        target_names = [target_name_with_node[0] for target_name_with_node in target_names_with_nodes]
 
-        target_names = list()
-        for target in node.targets:
-            tgt_names = self._extract_assign_ids(target)
-            target_names.extend(tgt_names)
-        logger.debug(target_names)
+        logger.debug(f"Applying hints to '{target_names}'")
+
+        class_name = None
+        node_to_check = node
+        while hasattr(node_to_check, "parent"):
+            node_to_check = node_to_check.parent  # type: ignore
+            if isinstance(node_to_check, ast.ClassDef):
+                class_name = node_to_check.name
+                break
 
-        var_mask = [
+        var_mask_local_variables = [
             self.df[TraceData.CATEGORY].isin(
-                [TraceDataCategory.LOCAL_VARIABLE, TraceDataCategory.CLASS_MEMBER]
+                [TraceDataCategory.LOCAL_VARIABLE]
             ),
             self.df[TraceData.VARNAME].isin(target_names),
             self.df[TraceData.LINENO] == node.lineno,
+            self.df[TraceData.CLASS] == class_name,
         ]
 
-        node_vars = self.df[functools.reduce(operator.and_, var_mask)]
+        var_mask_class_members = [
+            self.df[TraceData.CATEGORY] == TraceDataCategory.CLASS_MEMBER,
+            self.df[TraceData.VARNAME].isin(target_names),
+            self.df[TraceData.CLASS] == class_name,  # type: ignore
```

Comment:
why need to ignore?

Reply:
Must have accidentally placed it in the wrong place. Removed